openassessment/xblock/config_mixin.py

```diff
@@ -80,11 +81,14 @@ def is_feature_enabled(self, flag):
         Returns True if a CourseWaffleFlag, WaffleSwitch, or Django settings ``FEATURE``
         is enabled for this block, False otherwise.
         """
-        if hasattr(self, 'location') and self._course_waffle_flag(flag).is_enabled(self.location.course_key):
-            return True
+        try:
+            if hasattr(self, 'location') and self._course_waffle_flag(flag).is_enabled(self.location.course_key):
+                return True
 
-        if self._waffle_switch(flag).is_enabled():
-            return True
+            if self._waffle_switch(flag).is_enabled():
+                return True
+        except ImportError:
+            pass
```

Comment:
I tried that, but since the decorator is not wrapping anything, everything from the new decorator is run when OpenAssessmentBlock is imported. So, it is raising import error in the tests that
 has this line `from openassessment.xblock.openassessmentblock import OpenAssessmentBlock`. I could move these imports to the places where OpenAssessmentBlock is used in each test and mock that import_waffle_switch, but I was not sure if that was better than just this change. Let me know if you think that it would be better at that way.


On the other hand, I think that I cannot wrap in the decorator the part that uses the waffle switches because [the part](https://github.com/edx/edx-platform/blob/61e1eda20df2825a409db3e2d36c69d7c36d3e2d/lms/djangoapps/course_api/blocks/transformers/student_view.py#L54) in the platform that checks if a Xblock has mobile support needs that  xblocks.support('multi_device') is ran on import time.

Reply:
@morenol Ah thank you for explaining, I see now why it's complicated.

Could you move this `try`/`catch` block closer to the source of the issue, and add comments explaining? That way, it's clear that the exception is directly related to this feature:

```diff
--- a/openassessment/xblock/mobile.py
+++ b/openassessment/xblock/mobile.py
@@ -15,8 +15,12 @@ class TogglableMobileSupport(ConfigMixin):
         """
         Add mobile support if the feature flag is enabled.
         """
-        if self.is_mobile_support_waffle_enabled:
-            return XBlock.supports('multi_device')(view)
+        try:
+            if self.is_mobile_support_waffle_enabled:
+                return XBlock.supports('multi_device')(view)
+        except ImportError:
+            # openedx environment not available, e.g. in testing
+            pass
         return view
```